packages/medusa/src/services/price-list.ts

```diff
@@ -253,9 +253,21 @@ class PriceListService extends BaseService {
     config: FindConfig<PriceList> = { skip: 0, take: 20 }
   ): Promise<[PriceList[], number]> {
     const priceListRepo = this.manager_.getCustomRepository(this.priceListRepo_)
-
-    const query = this.buildQuery_(selector, config)
-    return await priceListRepo.findAndCount(query)
+    const q = selector.q
+    const { relations, ...query } = this.buildQuery_(selector, config)
+
+    if (q) {
+      delete query.where.q
```

Comment:
`query.where.q = undefined` ? for the same reasons describe earlier

Reply:
From a perf point of view, setting a property to undefined is more performant than deleting the property which mutate the object structure 😅
Dont know if it make sense. Here some explanation https://levelup.gitconnected.com/5-facts-about-delete-operator-in-javascript-c16fd2588cd?gi=4c1668543b9a

But in the end it really depends on the context so the choice is up to you since it is a micro opti here 😂

Reply:
Interesting! Thanks for sharing, I didn't know that 😄 

Personally, if perf is not a problem currently, I prefer the `delete` operator mostly because then we don't have to worry about the property lurking in case someone uses the object in a `for .. in` loop, but you might have a counter argument for this one, so please feel free to disagree and share!